applications/StructuralMechanicsApplication/custom_constitutive/linear_elastic_orthotropic_2D_law.cpp

```diff
@@ -0,0 +1,241 @@
+
+			//4.-Right Cauchy Green
+			Matrix RightCauchyGreen = prod(trans(DeformationGradientF), DeformationGradientF);
+
+			//5.-Green-Lagrange Strain:
+
+			//E= 0.5*(FT*F-1)
+			this->CalculateGreenLagrangeStrain(RightCauchyGreen, StrainVector);
+		}
+
+		//7.-Calculate Total PK2 stress
+
+		if (Options.Is(ConstitutiveLaw::COMPUTE_STRESS))
+		{
+			if (Options.Is(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR))
+			{
+				Matrix& ConstitutiveMatrix = rValues.GetConstitutiveMatrix();
+				this->CalculateLinearElasticMatrix(ConstitutiveMatrix, MaterialProperties);
+				this->CalculateStress(StrainVector, ConstitutiveMatrix, StressVector);
+			}
+			else {
+				Matrix ConstitutiveMatrix(StrainVector.size(), StrainVector.size());
+				noalias(ConstitutiveMatrix) = ZeroMatrix(StrainVector.size(), StrainVector.size());
+
+				this->CalculateLinearElasticMatrix(ConstitutiveMatrix, MaterialProperties);
+				this->CalculateStress(StrainVector, ConstitutiveMatrix, StressVector);
+			}
+		}
+		else if (Options.IsNot(ConstitutiveLaw::COMPUTE_STRESS) && Options.Is(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR))
+		{
+			Matrix& ConstitutiveMatrix = rValues.GetConstitutiveMatrix();
+			this->CalculateLinearElasticMatrix(ConstitutiveMatrix, MaterialProperties);
+		}
+	}
+	
+	//***********************COMPUTE TOTAL STRAIN*****************************************
+	//************************************************************************************
+
+	void LinearElasticOrthotropic2DLaw::CalculateGreenLagrangeStrain(const Matrix & rRightCauchyGreen,
+		Vector& rStrainVector)
+	{
+		// TAKEN FROM LinearElasticPlasticPlaneStrain2DLaw
+		//E= 0.5*(FT*F-1)
+		rStrainVector[0] = 0.5 * (rRightCauchyGreen(0, 0) - 1.00);
+		rStrainVector[1] = 0.5 * (rRightCauchyGreen(1, 1) - 1.00);
+		rStrainVector[2] = rRightCauchyGreen(0, 1);
+	}
+
+	//***********************COMPUTE TOTAL STRESS PK2*************************************
+	//************************************************************************************
+
+	void LinearElasticOrthotropic2DLaw::CalculateStress(const Vector & rStrainVector,
+		const Matrix & rConstitutiveMatrix,
+		Vector& rStressVector)
+	{
+		//1.-2nd Piola Kirchhoff StressVector increment
+		if (rStressVector.size() != rStrainVector.size())
+			rStressVector.resize(rStrainVector.size(), false);
+
+		noalias(rStressVector) = prod(rConstitutiveMatrix, rStrainVector);
+	}
+
+	//***********************COMPUTE LINEAR ELASTIC MATRIX**********************
+	//**************************************************************************
+
+	void LinearElasticOrthotropic2DLaw::CalculateLinearElasticMatrix(Matrix& rConstitutiveMatrix,
+		const Properties& rMaterialProperties)
+	{
+		//double G13 = G12;	// currently handled through "shell_cross_section.cpp"
+		//double G23 = G12;	// currently handled through "shell_cross_section.cpp"
+
+		double v21 = rMaterialProperties[POISSON_RATIO_XY]*rMaterialProperties[YOUNG_MODULUS_Y] / rMaterialProperties[YOUNG_MODULUS_X];
```

Comment:
use a local variable for nu_xy, so to access the database only once.

also make all of those local vars const